### PR TITLE
[TechDocs] Rebuild docs if the github source is newer than last docs build

### LINKED ISSUE #2330: TechDocs: Automatically rebuild documentation on updates

### PR DESCRIPTION
**Pull request by hooloovooo**
## Hey, I just made a Pull Request!

Rebuilds docs if it's github source has been updated.

closes: #2330 

#### :heavy_check_mark: Checklist

- [ ] All tests are passing `yarn test`
- [ ] Screenshots attached (for UI changes)
- [ ] Relevant documentation updated
- [ ] Prettier run on changed files
- [ ] Tests added for new functionality
- [ ] Regression tests added for bug fixes

### CODE REVIEW

File: docs/features/techdocs/getting-started.md [hunk 1]
```diff
@@ -74,17 +74,28 @@ export { plugin as TechDocs } from '@backstage/plugin-techdocs';
 ### Setting the configuration
 
 TechDocs allows for configuration of the docs storage URL through your
-`app-config` file.
+`app-config` file. We provide two different values to be configured,
+`requestUrl` and `storageUrl`. The `requestUrl` is what the reader will request
+it's data from, and `storageUrl` is where the backend can find the stored
```
Comment (freben) [inline docs/features/techdocs/getting-started.md:79]: its

File: packages/create-app/templates/default-app/app-config.yaml.hbs [hunk 2]
```diff
@@ -39,6 +39,8 @@ proxy:
 
 techdocs:
   storageUrl: http://localhost:7000/techdocs/static/docs
+  requestUrl: http://localhost:7000/techdocs/docs
+
```
Comment (freben) [inline packages/create-app/templates/default-app/app-config.yaml.hbs:43]: extra newline

**Review by freben — APPROVED** [review]
approved for default-branch - seems convenient, but just three weekly downloads? maybe there's some more well supported lib, perhaps something even from github themselves or something?

File: packages/create-app/templates/default-app/app-config.yaml.hbs (hunk 2 above)
Comment (emmaindal) [inline packages/create-app/templates/default-app/app-config.yaml.hbs:43]:
```suggestion
```

File: docs/features/techdocs/getting-started.md (hunk 1 above)
Comment (emmaindal) [inline docs/features/techdocs/getting-started.md:79]:
```suggestion
its data from, and `storageUrl` is where the backend can find the stored
```

**Comment by emmaindal** [comment]
> approved for default-branch - seems convenient, but just three weekly downloads? maybe there's some more well supported lib, perhaps something even from github themselves or something?

@freben good point. we will take a look!

**Comment by hooloovooo** [comment]
> approved for default-branch - seems convenient, but just three weekly downloads? maybe there's some more well supported lib, perhaps something even from github themselves or something?

Couldn't find a good alternative. It is a very very lightweight library without dependencies though, only downside is that it doesn't have type definitions. I'd be open to writing our own version of it though since it would probably only be a few lines of code.

**Comment by emmaindal** [comment]
> > approved for default-branch - seems convenient, but just three weekly downloads? maybe there's some more well supported lib, perhaps something even from github themselves or something?
> 
> Couldn't find a good alternative. It is a very very lightweight library without dependencies though, only downside is that it doesn't have type definitions. I'd be open to writing our own version of it though since it would probably only be a few lines of code.

Yeah I was thinking about that as well when looking into what the lib actually did. I think we could just write our own little helper func

**Review by timja — APPROVED** [review]
it works

**Review by emmaindal — APPROVED** [review]
I think we can go with default-branch lib for now and create our own utility function in a separate PR?

**Pull request merged**